src/AEG/AEGTree.ts

```diff
@@ -224,6 +239,24 @@ export class AEGTree {
         }
     }
 
+    /**
+     * Method that checks if an AEG Tree is equal to another AEG Tree. Trees are equal if they have
+     * the same children, irrespective of the ordering of nodes within a level
+     * @param otherTree The tree we want to check for equality with
+     * @returns True, if the trees are equal. Else, false
+     */
+    public isEqualTo(otherTree: AEGTree): boolean {
+        //For 2 trees to be equal, they must have the same number of children
+        if (this.sheet.children.length === otherTree.sheet.children.length) {
+            if (this.sheet.children.length === 0) {
+                //Both trees are empty trees => they are equal
+                return true;
+            }
+            return this.sheet.isEqualTo(otherTree.sheet);
+        }
+        return false;
```

Comment:
Since you run this exact same check for child length in `CutNode.isEqualTo`, there is no need to run it here since it only saves us a single function call, a mere constant factor.  
```suggestion
        return this.sheet.isEqualTo(otherTree.sheet);
```